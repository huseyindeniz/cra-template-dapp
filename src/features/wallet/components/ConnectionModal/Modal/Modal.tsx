import {
  Modal as ChakraModal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalCloseButton,
  ModalBody,
  Divider,
} from '@chakra-ui/react';
import { FaFileSignature } from '@react-icons/all-files/fa/FaFileSignature';
import { GiChoice } from '@react-icons/all-files/gi/GiChoice';
import { IoMdUnlock } from '@react-icons/all-files/io/IoMdUnlock';
import { MdExtension } from '@react-icons/all-files/md/MdExtension';
import { Step, Steps } from 'chakra-ui-steps';
import React from 'react';
import { useTranslation } from 'react-i18next';

export interface ModalProps {
  isOpen: boolean;
  activeStep: number;
  stepState?: 'loading' | 'error';
  checkWalletContent?: React.ReactNode;
  checkAccountContent?: React.ReactNode;
  checkNetworkContent?: React.ReactNode;
  checkSignContent?: React.ReactNode;
  orientation?: 'horizontal' | 'vertical'; // this is a temp fix for unit tests. It should always be "vertical"
  onDisconnect: () => void;
}

export const Modal: React.FC<ModalProps> = ({
  isOpen,
  activeStep,
  stepState,
  checkWalletContent,
  checkAccountContent,
  checkNetworkContent,
  checkSignContent,
  orientation = 'vertical',
  onDisconnect,
}) => {
  const { t } = useTranslation('FeatureWallet');

  const connectionSteps = (
    <Steps
      orientation={orientation}
      activeStep={activeStep}
      state={stepState}
      colorScheme="whatsapp"
      responsive={false}
    >
      <Step
        width="100%"
        label={t('Check Metamask Extension')}
        key="installation"
        description={
          t('The Metamask wallet extension needs to be installed.') as string
        }
        icon={MdExtension}
      >
        {checkWalletContent}
      </Step>
      <Step
        width="100%"
        label={t('Check Metamask Status')}
        key="unlock"
        description={t('The Metamask wallet needs to be unlocked.') as string}
        icon={IoMdUnlock}
      >
        {checkAccountContent}
      </Step>
      <Step
        width="100%"
        label={t('Check Metamask Network')}
        key="network"
        description={
          t(
            'A supported network needs to be selected in the Metamask wallet.'
          ) as string
        }
        icon={GiChoice}
      >
        {checkNetworkContent}
      </Step>
      <Step
        width="100%"
        label={t('Check Metamask Signature')}
        key="signin"
        description={
          t(
            'The login request needs to be signed in the Metamask wallet.'
          ) as string
        }
        icon={FaFileSignature}
      >
        {checkSignContent}
      </Step>
    </Steps>
  );

  return (
    <ChakraModal
      closeOnOverlayClick={false}
      isOpen={isOpen}
      onClose={() => onDisconnect()}
      size="xl"
      preserveScrollBarGap={true}
    >
      <ModalOverlay />
      <ModalContent>
        <ModalHeader>
          {t('Connecting to Metamask')}:
          <Divider mt={1} />
        </ModalHeader>
        <ModalCloseButton />
        <ModalBody mb={3}>{connectionSteps}</ModalBody>
      </ModalContent>
    </ChakraModal>
  );
};
