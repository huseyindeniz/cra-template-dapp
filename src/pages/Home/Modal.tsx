import React from "react";
import { useTranslation } from "react-i18next";
import { Link as RouterLink } from "react-router-dom";
import {
  Modal as ChakraModal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalCloseButton,
  ModalBody,
  Divider,
  Button,
  useDisclosure,
  Center,
} from "@chakra-ui/react";
import ReactPlayer from "react-player";

export const Modal: React.FC = () => {
  const { t, i18n } = useTranslation("PageHome");
  const { onOpen, onClose, isOpen } = useDisclosure();
  return (
    <>
      <Button
        onClick={onOpen}
        colorScheme={"blue"}
        bg={"blue.400"}
        rounded={"full"}
        px={6}
        _hover={{
          bg: "blue.500",
        }}
      >
        {t("Get Started")}
      </Button>
      <ChakraModal
        closeOnOverlayClick={false}
        onClose={onClose}
        isOpen={isOpen}
        size={"xl"}
        preserveScrollBarGap={true}
      >
        <ModalOverlay />
        <ModalContent>
          <ModalHeader>
            {t("Get Started")}:
            <Divider mt={1} />
          </ModalHeader>
          <ModalCloseButton />
          <ModalBody mb={3}>
            <ReactPlayer
              width={"100%"}
              url="https://www.youtube.com/watch?v=SWB2PnDzrUc"
              controls={true}
              light={"./assets/images/get-started.png"}
            />
            <Center m={2} p={2}>
              <Button
                as={RouterLink}
                to={"/" + i18n.language + "/documentation"}
                variant={"link"}
                colorScheme={"blue"}
                size={"sm"}
              >
                {t("See full documentation")}
              </Button>
            </Center>
          </ModalBody>
        </ModalContent>
      </ChakraModal>
    </>
  );
};
