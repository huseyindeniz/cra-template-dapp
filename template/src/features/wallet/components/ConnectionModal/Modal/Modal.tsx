import {
  Modal as ChakraModal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalCloseButton,
  ModalBody,
  Divider,
  Box,
  Step,
  StepDescription,
  StepIcon,
  StepIndicator,
  StepSeparator,
  StepStatus,
  StepTitle,
  Stepper,
  Spinner,
  Container,
} from '@chakra-ui/react';
import { FaFileSignature } from '@react-icons/all-files/fa/FaFileSignature';
import { GiChoice } from '@react-icons/all-files/gi/GiChoice';
import { IoMdUnlock } from '@react-icons/all-files/io/IoMdUnlock';
import { MdAccountBalanceWallet } from '@react-icons/all-files/md/MdAccountBalanceWallet';
import { MdError } from '@react-icons/all-files/md/MdError';
import { MdExtension } from '@react-icons/all-files/md/MdExtension';
import React from 'react';
import { useTranslation } from 'react-i18next';

import { DISABLE_WALLET_SIGN } from '../../../config';

export interface ModalProps {
  isOpen: boolean;
  activeStep: number;
  stepState?: 'loading' | 'error';
  checkWalletContent?: React.ReactNode;
  checkAccountContent?: React.ReactNode;
  checkNetworkContent?: React.ReactNode;
  checkSignContent?: React.ReactNode;
  onDisconnect: () => void;
}

export const Modal: React.FC<ModalProps> = ({
  isOpen,
  activeStep,
  stepState,
  checkWalletContent,
  checkAccountContent,
  checkNetworkContent,
  checkSignContent,
  onDisconnect,
}) => {
  const { t } = useTranslation('FeatureWallet');

  const connectionSteps = (
    <Stepper index={activeStep} orientation="vertical" gap={'40px'}>
      <Step key="installation">
        <StepIndicator>
          <StepStatus
            complete={<StepIcon />}
            incomplete={<MdExtension />}
            active={
              stepState !== undefined ? (
                stepState === 'error' ? (
                  <MdError />
                ) : (
                  <Spinner />
                )
              ) : (
                <MdExtension />
              )
            }
          />
        </StepIndicator>
        <Box>
          <StepTitle>{t('Check Metamask Extension')}</StepTitle>
          <StepDescription>
            {
              t(
                'The Metamask wallet extension needs to be installed.'
              ) as string
            }
          </StepDescription>
          <Container my={2} p={0} centerContent>
            {checkWalletContent}
          </Container>
        </Box>
        <StepSeparator />
      </Step>
      <Step key="unlock">
        <StepIndicator>
          <StepStatus
            complete={<StepIcon />}
            incomplete={<IoMdUnlock />}
            active={
              stepState !== undefined ? (
                stepState === 'error' ? (
                  <MdError />
                ) : (
                  <Spinner />
                )
              ) : (
                <IoMdUnlock />
              )
            }
          />
        </StepIndicator>
        <Box>
          <StepTitle>{t('Check Metamask Status')}</StepTitle>
          <StepDescription>
            {t('The Metamask wallet needs to be unlocked.') as string}
          </StepDescription>
          <Container my={2} p={0} centerContent>
            {checkAccountContent}
          </Container>
        </Box>
        <StepSeparator />
      </Step>
      <Step key="network">
        <StepIndicator>
          <StepStatus
            complete={<StepIcon />}
            incomplete={<GiChoice />}
            active={
              stepState !== undefined ? (
                stepState === 'error' ? (
                  <MdError />
                ) : (
                  <Spinner />
                )
              ) : (
                <GiChoice />
              )
            }
          />
        </StepIndicator>
        <Box>
          <StepTitle>{t('Check Metamask Network')}</StepTitle>
          <StepDescription>
            {
              t(
                'A supported network needs to be selected in the Metamask wallet.'
              ) as string
            }
          </StepDescription>
          <Container my={2} p={0} centerContent>
            {checkNetworkContent}
          </Container>
        </Box>
        <StepSeparator />
      </Step>
      <Step key="signin">
        <StepIndicator>
          <StepStatus
            complete={<StepIcon />}
            incomplete={
              DISABLE_WALLET_SIGN ? (
                <MdAccountBalanceWallet />
              ) : (
                <FaFileSignature />
              )
            }
            active={
              stepState !== undefined ? (
                stepState === 'error' ? (
                  <MdError />
                ) : (
                  <Spinner />
                )
              ) : DISABLE_WALLET_SIGN ? (
                <MdAccountBalanceWallet />
              ) : (
                <FaFileSignature />
              )
            }
          />
        </StepIndicator>
        <Box>
          <StepTitle>
            {DISABLE_WALLET_SIGN
              ? t('Load Wallet Account')
              : t('Check Metamask Signature')}
          </StepTitle>
          <StepDescription>
            {DISABLE_WALLET_SIGN ??
              t('The login request needs to be signed in the Metamask wallet.')}
          </StepDescription>
          <Container my={2} p={0} centerContent>
            {checkSignContent}
          </Container>
        </Box>
        <StepSeparator />
      </Step>
    </Stepper>
  );

  return (
    <ChakraModal
      closeOnOverlayClick={false}
      isOpen={isOpen}
      onClose={() => onDisconnect()}
      preserveScrollBarGap={true}
    >
      <ModalOverlay />
      <ModalContent mx={4} maxWidth={{ base: '90%', md: '600px' }}>
        <ModalHeader>
          {t('Connecting to Metamask')}:
          <Divider mt={1} />
        </ModalHeader>
        <ModalCloseButton />
        <ModalBody mb={3}>{connectionSteps}</ModalBody>
      </ModalContent>
    </ChakraModal>
  );
};
