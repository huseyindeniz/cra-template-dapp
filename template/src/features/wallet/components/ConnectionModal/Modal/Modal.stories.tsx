import { ComponentStory, ComponentMeta } from '@storybook/react';
import { composeStories } from '@storybook/testing-react';

import * as checkAccountStories from '../Steps/CheckAccount.stories';
import * as checkNetworkStories from '../Steps/CheckNetwork.stories';
import * as checkSignStories from '../Steps/CheckSign.stories';
import * as checkWalletStories from '../Steps/CheckWallet.stories';

import { Modal } from './Modal';

export default {
  title: 'wallet/ConnectionModal/Modal',
  component: Modal,
} as ComponentMeta<typeof Modal>;

// STEP: CHECK WALLET

const {
  CheckWalletIdle,
  InitRequested,
  InitFailed,
  NotSupported,
  Initialized,
} = composeStories(checkWalletStories);

const TemplateCheckWallet: ComponentStory<typeof Modal> = args => (
  <Modal {...args} isOpen={true} activeStep={0} />
);

export const CheckWalletCheckWalletIdle = TemplateCheckWallet.bind({});
CheckWalletCheckWalletIdle.args = {
  checkWalletContent: <CheckWalletIdle />,
};

export const CheckWalletInitRequested = TemplateCheckWallet.bind({});
CheckWalletInitRequested.args = {
  stepState: 'loading',
  checkWalletContent: <InitRequested />,
};

export const CheckWalletInitFailed = TemplateCheckWallet.bind({});
CheckWalletInitFailed.args = {
  stepState: 'error',
  checkWalletContent: <InitFailed />,
};

export const CheckWalletNotSupported = TemplateCheckWallet.bind({});
CheckWalletNotSupported.args = {
  stepState: 'error',
  checkWalletContent: <NotSupported />,
};

export const CheckWalletInitialized = TemplateCheckWallet.bind({});
CheckWalletInitialized.args = {
  checkWalletContent: <Initialized />,
};

// STEP: CHECK ACCOUNT

const {
  CheckAccountIdle,
  AccountRequested,
  AccountDetectionFailed,
  Locked,
  UnlockRequested,
  UnlockRejected,
  UnlockFailed,
  AccountLoaded,
} = composeStories(checkAccountStories);

const TemplateCheckAccount: ComponentStory<typeof Modal> = args => (
  <Modal {...args} isOpen={true} activeStep={1} />
);

export const CheckAccountCheckAccountIdle = TemplateCheckAccount.bind({});
CheckAccountCheckAccountIdle.args = {
  checkAccountContent: <CheckAccountIdle />,
};

export const CheckAccountAccountRequested = TemplateCheckAccount.bind({});
CheckAccountAccountRequested.args = {
  stepState: 'loading',
  checkAccountContent: <AccountRequested />,
};

export const CheckAccountAccountDetectionFailed = TemplateCheckAccount.bind({});
CheckAccountAccountDetectionFailed.args = {
  stepState: 'error',
  checkAccountContent: <AccountDetectionFailed />,
};

export const CheckAccountLocked = TemplateCheckAccount.bind({});
CheckAccountLocked.args = {
  stepState: 'error',
  checkAccountContent: <Locked />,
};

export const CheckAccountUnlockRequested = TemplateCheckAccount.bind({});
CheckAccountUnlockRequested.args = {
  stepState: 'loading',
  checkAccountContent: <UnlockRequested />,
};

export const CheckAccountUnlockRejected = TemplateCheckAccount.bind({});
CheckAccountUnlockRejected.args = {
  stepState: 'error',
  checkAccountContent: <UnlockRejected />,
};

export const CheckAccountUnlockFailed = TemplateCheckAccount.bind({});
CheckAccountUnlockFailed.args = {
  stepState: 'error',
  checkAccountContent: <UnlockFailed />,
};

export const CheckAccountAccountLoaded = TemplateCheckAccount.bind({});
CheckAccountAccountLoaded.args = {
  checkAccountContent: <AccountLoaded />,
};

// STEP: CHECK NETWORK

const {
  CheckNetworkIdle,
  NetworkRequested,
  NetworkDetectionFailed,
  WrongNetwork,
  NetworkSwitchRequested,
  NetworkSwitchRejected,
  NetworkSwitchFailed,
  NetworkLoaded,
} = composeStories(checkNetworkStories);

const TemplateCheckNetwork: ComponentStory<typeof Modal> = args => (
  <Modal {...args} isOpen={true} activeStep={2} />
);

export const CheckNetworkCheckNetworkIdle = TemplateCheckNetwork.bind({});
CheckNetworkCheckNetworkIdle.args = {
  checkNetworkContent: <CheckNetworkIdle />,
};

export const CheckNetworkNetworkRequested = TemplateCheckNetwork.bind({});
CheckNetworkNetworkRequested.args = {
  stepState: 'loading',
  checkNetworkContent: <NetworkRequested />,
};

export const CheckNetworkNetworkDetectionFailed = TemplateCheckNetwork.bind({});
CheckNetworkNetworkDetectionFailed.args = {
  stepState: 'error',
  checkNetworkContent: <NetworkDetectionFailed />,
};

export const CheckNetworkWrongNetwork = TemplateCheckNetwork.bind({});
CheckNetworkWrongNetwork.args = {
  stepState: 'error',
  checkNetworkContent: <WrongNetwork />,
};

export const CheckNetworkSwitchRequested = TemplateCheckNetwork.bind({});
CheckNetworkSwitchRequested.args = {
  stepState: 'loading',
  checkNetworkContent: <NetworkSwitchRequested />,
};

export const CheckNetworkSwitchRejected = TemplateCheckNetwork.bind({});
CheckNetworkSwitchRejected.args = {
  stepState: 'error',
  checkNetworkContent: <NetworkSwitchRejected />,
};

export const CheckNetworkSwitchFailed = TemplateCheckNetwork.bind({});
CheckNetworkSwitchFailed.args = {
  stepState: 'error',
  checkNetworkContent: <NetworkSwitchFailed />,
};

export const CheckNetworkLoaded = TemplateCheckNetwork.bind({});
CheckNetworkLoaded.args = {
  checkNetworkContent: <NetworkLoaded />,
};

// STEP: CHECK SIGN

const {
  CheckSignIdle,
  NotSigned,
  SignRequested,
  SignRejected,
  SignTimedOut,
  SignFailed,
  Signed,
} = composeStories(checkSignStories);

const TemplateCheckSign: ComponentStory<typeof Modal> = args => (
  <Modal {...args} isOpen={true} activeStep={3} />
);

export const CheckSignCheckSignIdle = TemplateCheckSign.bind({});
CheckSignCheckSignIdle.args = {
  checkSignContent: <CheckSignIdle />,
};

export const CheckSignNotSigned = TemplateCheckSign.bind({});
CheckSignNotSigned.args = {
  stepState: 'error',
  checkSignContent: <NotSigned />,
};

export const CheckSignSignRequested = TemplateCheckSign.bind({});
CheckSignSignRequested.args = {
  stepState: 'loading',
  checkSignContent: <SignRequested />,
};

export const CheckSignSignRejected = TemplateCheckSign.bind({});
CheckSignSignRejected.args = {
  stepState: 'error',
  checkSignContent: <SignRejected />,
};

export const CheckSignSignTimedOut = TemplateCheckSign.bind({});
CheckSignSignTimedOut.args = {
  stepState: 'error',
  checkSignContent: <SignTimedOut />,
};

export const CheckSignSignFailed = TemplateCheckSign.bind({});
CheckSignSignFailed.args = {
  stepState: 'error',
  checkSignContent: <SignFailed />,
};

export const CheckSignSigned = TemplateCheckSign.bind({});
CheckSignSigned.args = {
  checkSignContent: <Signed />,
};
