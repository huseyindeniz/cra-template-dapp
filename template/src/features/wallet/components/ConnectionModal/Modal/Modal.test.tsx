import { render } from '@testing-library/react';

import { Modal } from './Modal';

describe.skip('Feature: Wallet', () => {
  describe('Component: ConnectionModal/Modal', () => {
    const onClickSpy = jest.fn();

    describe('Scenario: isOpen=false', () => {
      it('should be empty', () => {
        // Arrange
        // Act
        const result1 = render(
          <Modal activeStep={0} isOpen={false} onDisconnect={onClickSpy} />
        );
        const result2 = render(
          <Modal
            activeStep={0}
            isOpen={false}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        expect(result1.container.childElementCount).toEqual(0);
        expect(result2.container.childElementCount).toEqual(0);
      });
    });

    describe('Scenario: OnDisconnect', () => {
      it('when modal closed disconnet should be called', () => {
        // Arrange
        // Act
        const { getByRole } = render(
          <Modal
            activeStep={0}
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        getByRole('button').click();
        // Assert
        expect(onClickSpy).toHaveBeenCalled();
      });
    });

    describe('Scenario: CheckWallet', () => {
      const stateTitle = 'Check Metamask Extension';
      const stateDesc = 'The Metamask wallet extension needs to be installed.';
      const stateContent = 'My Mock Check Wallet Step Content';

      it('when stepState is undefined it should show plain step logo', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={0}
            checkWalletContent={stateContent}
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check plain step icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });

      it('when stepState is error it should show error icon', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={0}
            checkWalletContent={stateContent}
            stepState="error"
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check error icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });

      // Loading state is failing, probably a bug in Chakra-Steps package
      it.skip('when stepState is loading it should show loading icon', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={0}
            checkWalletContent={stateContent}
            stepState="loading"
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check loading icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });
    });

    describe('Scenario: CheckAccount', () => {
      const stateTitle = 'Check Metamask Status';
      const stateDesc = 'The Metamask wallet needs to be unlocked.';
      const stateContent = 'My Mock Check Account Step Content';

      it('when stepState is undefined it should show plain step logo', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={1}
            checkAccountContent={stateContent}
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check plain step icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });

      it('when stepState is error it should show error icon', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={1}
            checkAccountContent={stateContent}
            stepState="error"
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check error icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });

      // Loading state is failing, probably a bug in Chakra-Steps package
      it.skip('when stepState is loading it should show loading icon', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={1}
            checkAccountContent={stateContent}
            stepState="loading"
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check loading icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });
    });

    describe('Scenario: CheckNetwork', () => {
      const stateTitle = 'Check Metamask Network';
      const stateDesc =
        'A supported network needs to be selected in the Metamask wallet.';
      const stateContent = 'My Mock Check Network Step Content';

      it('when stepState is undefined it should show plain step logo', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={2}
            checkNetworkContent={stateContent}
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check plain step icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });

      it('when stepState is error it should show error icon', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={2}
            checkNetworkContent={stateContent}
            stepState="error"
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check error icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });

      // Loading state is failing, probably a bug in Chakra-Steps package
      it.skip('when stepState is loading it should show loading icon', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={2}
            checkNetworkContent={stateContent}
            stepState="loading"
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check loading icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });
    });

    describe('Scenario: CheckSign', () => {
      const stateTitle = 'Check Metamask Signature';
      const stateDesc =
        'The login request needs to be signed in the Metamask wallet.';
      const stateContent = 'My Mock Check Sign Step Content';

      it('when stepState is undefined it should show plain step logo', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={3}
            checkSignContent={stateContent}
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check plain step icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });

      it('when stepState is error it should show error icon', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={3}
            checkSignContent={stateContent}
            stepState="error"
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check error icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });

      // Loading state is failing, probably a bug in Chakra-Steps package
      it.skip('when stepState is loading it should show loading icon', () => {
        // Arrange
        // Act
        const { asFragment, getByText } = render(
          <Modal
            activeStep={3}
            checkSignContent={stateContent}
            stepState="loading"
            isOpen={true}
            onDisconnect={onClickSpy}
            orientation="horizontal"
          />
        );
        // Assert
        const title = getByText(new RegExp(stateTitle, 'i'));
        const desc = getByText(stateDesc);
        // TODO: check loading icon
        expect(title).toBeTruthy();
        expect(desc).toBeTruthy();
        expect(asFragment).toMatchSnapshot();
      });
    });
  });
});
