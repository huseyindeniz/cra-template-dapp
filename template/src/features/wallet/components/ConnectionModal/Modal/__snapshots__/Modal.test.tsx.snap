// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: Wallet Component: ConnectionModal/Modal Scenario: CheckAccount when stepState is error it should show error icon 1`] = `[Function]`;

exports[`Feature: Wallet Component: ConnectionModal/Modal Scenario: CheckAccount when stepState is undefined it should show plain step logo 1`] = `[Function]`;

exports[`Feature: Wallet Component: ConnectionModal/Modal Scenario: CheckNetwork when stepState is error it should show error icon 1`] = `[Function]`;

exports[`Feature: Wallet Component: ConnectionModal/Modal Scenario: CheckNetwork when stepState is undefined it should show plain step logo 1`] = `[Function]`;

exports[`Feature: Wallet Component: ConnectionModal/Modal Scenario: CheckSign when stepState is error it should show error icon 1`] = `[Function]`;

exports[`Feature: Wallet Component: ConnectionModal/Modal Scenario: CheckSign when stepState is undefined it should show plain step logo 1`] = `[Function]`;

exports[`Feature: Wallet Component: ConnectionModal/Modal Scenario: CheckWallet when stepState is error it should show error icon 1`] = `[Function]`;

exports[`Feature: Wallet Component: ConnectionModal/Modal Scenario: CheckWallet when stepState is loading it should show loading icon 1`] = `[Function]`;

exports[`Feature: Wallet Component: ConnectionModal/Modal Scenario: CheckWallet when stepState is undefined it should show plain step logo 1`] = `[Function]`;
